src/lang-promql/promql.ts

```diff
@@ -0,0 +1,28 @@
+import { LezerSyntax } from "@codemirror/next/syntax";
+import { parser } from "lezer-promql";
+import { styleTags } from "@codemirror/next/highlight";
+import { Extension } from "@codemirror/next/state";
+
+export const promQLSyntax = new LezerSyntax(parser.withProps(
+  styleTags({
+    LineComment: "comment",
+    LabelName: "name",
+    StringLiteral: "string",
+    NumberLiteral: "number",
+    Duration: "number",
+    "abs absent absent_over_time avg_over_time ceil changes clamp_max clamp_min count_over_time days_in_month day_of_month day_of_week delta deriv exp floor histogram_quantile holt_winters hour idelta increase irate label_replace label_join ln log10 log2 max_over_time min_over_time minute month predict_linear quantile_over_time rate resets round scalar sort sort_desc sqrt stddev_over_time stdvar_over_time sum_over_time time timestamp vector year": "keyword",
+    "avg bottomk count count_values max min quantile stddev stdvar sum topk": "keyword definition",
+    "group_left group_right by without bool offset on and or unless": "operator",
```

Comment:
I split these into:

```
      'by without bool on ignoring group_left group_right offset': 'modifier',
      'and unless or': 'keyword',
```

Although the latter could also be `operator`.

Reply:
I think `by without bool on ignoring group_left group_right offset` should have a different color than `sum avg, count`. 
That's also why I put it as operator for both of them.

What do you think ?

Reply:
Hmm, they are definitely not operators though, they are modifiers (even if `modifier` is just mapped to `keyword` in https://codemirror.net/6/docs/ref/#highlight). I would also tend to give them the same color as the aggregators (which I would classify as keywords), but maybe that's just me...

Reply:
Btw. are we sure that sub-types aren't separately styleable? Because the highlight module docs say, "Types may extend other types. When no style for a given tag is present in the highlighter, it will fall back first to styles for that type without one or more flags, and then, if that also fails, to its parent types."

Reply:
mmm ok you got a point here. You are right in a sense where I shouldn't mixe theming and types styleable.

So we should go like you suggest. And then we will treat the theme in a different topic.